tests/lib/tools/nested-state

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: prepare"
```

Comment:
What do prepare and restore do, conceptually? We have a bunch of sub-commands for starting and stopping specific things and a generic prepare/restore. Does it make sense to have them all together?

Could you add a paragraph or two about what those commands to just beneath the usage output? Terse usage is great when there is little complexity and usage is obvious. When sub-commands multiply it is only good to tell you there's stuff you need to read about elsewhere.

Reply:
Added doc for all the commands